src/kernel/filesystem/fat32.zig

```diff
@@ -1195,16 +1216,56 @@ pub fn Fat32FS(comptime StreamType: type) type {
         /// Return: *vfs.Node
         ///     The VFS Node for the opened file/folder.
         ///
-        /// Error: Allocator.Error || ReadError || SeekError || vfs.Error
+        /// Error: Allocator.Error || vfs.Error
         ///     Allocator.Error           - Not enough memory for allocating memory
-        ///     ReadError                 - Errors when reading the stream.
-        ///     SeekError                 - Errors when seeking the stream.
         ///     vfs.Error.NoSuchFileOrDir - Error if the file/folder doesn't exist.
+        ///     vfs.Error.Unexpected      - An error occurred whilst reading the file system, this
+        ///                                 can be caused by a parsing error or errors on reading
+        ///                                 or seeking the underlying stream. If this occurs, then
+        ///                                 the real error is printed using `log.err`.
         ///
-        fn openImpl(fs: *const vfs.FileSystem, dir: *const vfs.DirNode, name: []const u8) (Allocator.Error || ReadError || SeekError || vfs.Error)!*vfs.Node {
+        fn openImpl(fs: *const vfs.FileSystem, dir: *const vfs.DirNode, name: []const u8) (Allocator.Error || vfs.Error)!*vfs.Node {
             const self = @fieldParentPtr(Fat32Self, "instance", fs.instance);
+            // TODO: Cache the files in this dir, so when opening, don't have to iterator the directory every time
+
+            // Iterate over the directory and find the file/folder
+            const cluster = try self.getDirCluster(dir);
+            var it = EntryIterator.init(self.allocator, self.fat_config, cluster, self.stream) catch |e| switch (e) {
+                error.OutOfMemory => return error.OutOfMemory,
+                else => {
+                    log.err("Error initialising the entry iterator. Error: {}\n", .{e});
+                    return vfs.Error.Unexpected;
+                },
+            };
+            defer it.deinit();
+            while (it.next() catch |e| switch (e) {
+                error.OutOfMemory => return vfs.Error.IsAFile,
+                else => {
+                    log.err("Error initialising the entry iterator. Error: {}\n", .{e});
+                    return vfs.Error.Unexpected;
+                },
+            }) |entry| {
+                defer entry.deinit();
+                // File name compare is case insensitive
```

Comment:
I think file names should be case sensitive, just like on Linux.

Reply:
FAT32 specifically says that they are case insensitive. but Ext2/3/4 is case sensitive so depends on the filesystem

Reply:
Oh I see, I didn't realise this sort of thing was part of the spec.